src/protocol/mapped-location-cache.ts

```diff
@@ -0,0 +1,54 @@
+const { sendMessage } = require("./socket");
+const { defer } = require("./utils");
+
+interface Location {
+  scriptId: string;
+  line: number;
+  column: number;
+}
+
+type MappedLocation = Location[];
+
+export class MappedLocationCache {
```

Comment:
This looks great. I wonder if down the road, we would want to make this a more generic feature of the protocol layer, where you could opt to make certain messages cacheable...